ambry-clustermap/src/main/java/com.github.ambry.clustermap/AmbryPartition.java

```diff
@@ -46,7 +47,7 @@
    * Instantiate an AmbryPartition instance.
    * @param id the id associated with this partition.
    * @param partitionClass the partition class that this partition belongs to
-   * @param clusterManagerCallback the {@link ClusterManagerCallback} to use to make callbacks
+   * @param clusterManagerCallback the {@link HelixClusterManager.HelixClusterManagerCallback} to use to make callbacks
    *                               to the {@link HelixClusterManager}
    * The initial state defaults to {@link PartitionState#READ_WRITE}.
    */
```

Comment:
And then pass in `ClusterManagerCallback<AmbryReplica, AmbryDisk, AmbryPartition>` here and in other similar classes, which will get rid of the need for the copies/casts.